chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -209,18 +217,34 @@
   case
   when ml2_mech_drivers.include?("openvswitch")
     # package is already installed
-
     neutron_agent = node[:neutron][:platform][:ovs_agent_name]
     agent_config_path = "/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini"
     interface_driver = "neutron.agent.linux.interface.OVSInterfaceDriver"
-    external_network_bridge = "br-public"
+    bridge_mappings = "floating:br-public"
+    if multiple_external_networks
+      bridge_mappings += ", "
+      bridge_mappings += neutron[:neutron][:additional_external_networks].collect { |n| n + ":" + "br-" + n }.join ','
+    end
+    if ml2_type_drivers.include?("vlan")
+      bridge_mappings += ", physnet1:br-fixed"
+    end
   when ml2_mech_drivers.include?("linuxbridge")
     package node[:neutron][:platform][:lb_agent_pkg]
 
     neutron_agent = node[:neutron][:platform][:lb_agent_name]
     agent_config_path = "/etc/neutron/plugins/linuxbridge/linuxbridge_conf.ini"
     interface_driver = "neutron.agent.linux.interface.BridgeInterfaceDriver"
-    external_network_bridge = ""
+    physnet = node[:crowbar_wall][:network][:nets][:nova_fixed].first rescue nil
+    interface_mappings = "physnet1:" + physnet
```

Comment:
the rescue nil and then +physnet in the line below doesn't look good to me. if this really can not happen, then remove the rescue nil (avoid hiding the real error). if it can happen, the rescue should return something more sensible.